src/status_im/utils/handlers_macro.cljs

```diff
@@ -1,16 +1,23 @@
 (ns status-im.utils.handlers-macro
-  (:require-macros status-im.utils.handlers-macro))
+  (:require-macros status-im.utils.handlers-macro)
+  (:require [clojure.set :as set]))
 
 (defn update-db [cofx fx]
   (if-let [db (:db fx)]
     (assoc cofx :db db)
     cofx))
 
+(def ^:private tx-keys #{:data-store/tx :data-store/base-tx})
+
 (defn safe-merge [fx new-fx]
   (if (:merging-fx-with-common-keys fx)
     fx
-    (let [common-keys (clojure.set/intersection (into #{} (keys fx))
-                                                (into #{} (keys new-fx)))]
-      (if (empty? (disj common-keys :db))
-        (merge fx new-fx)
+    (let [common-keys (set/intersection (into #{} (keys fx))
```

Comment:
Should we add some comments to this changes? Or probably to `merge-fx`'s doc string as it's behaviour is changed.